stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp

```diff
@@ -63,23 +72,43 @@ neg_binomial_2_log_glm_lpmf(const T_y& y, const T_x& x, const T_alpha& alpha,
   typedef
       typename stan::partials_return_type<T_y, T_x, T_alpha, T_beta,
                                           T_precision>::type T_partials_return;
+  typedef typename std::conditional<
+      is_vector<T_precision>::value,
+      typename std::conditional<
```

Comment:
Now that Im seeing more use of these, is there a way to pull out some kind of traits metaprograms that will accomplish this in a more readable way?  These nested versionare hard to read.  

The `stan::` qualifiers aren't necessary as we're in the `stan` namespace, but feel free to leave if you think it's clearer this way.

Reply:
I don't see a general way how to do it, as we have different types (Array/Matrix, sometimes even std::vector), and one version even depends on two different inputs. Except for nested one that is not actually needed, I will just leave that as it is for now.